pulser/simulation/simconfig.py

```diff
@@ -0,0 +1,129 @@
+        noisy simulations.
+
+        Keyword arguments:
+            noise_types (string list): types of noises to be used in the
+                simulation.
+            samples_per_run (int value): number of samples per noisy run.
+                Useful for cutting down on computing time, but unrealistic.
+            runs (int value): number of runs needed : each run
+                draws a new random noise.
+            laser_waist (float): waist of the gaussian laser in global pulses.
+            solver_options (qutip.Options): options for the qutip solver.
+        """
+    def __init__(self, noise_types: List[str] = [], runs: int = 15,
+                 samples_per_run: int = 5, temperature: float = 50.,
+                 laser_waist: float = 175.,
+                 solver_options: qutip.Options = qutip.Options(max_step=5)):
+        self.noise = noise_types
+        self.temperature = temperature
+        self.runs = runs
+        self.samples_per_run = samples_per_run
+        self.laser_waist = laser_waist
+        self.solver_options = solver_options
+        self.init_spam()
+
+    def __str__(self) -> str:
+        s = ""
+        s += "Options:\n"
+        s += "----------       \n"
+        s += "noise types:       " + str(self.noise) + "\n"
+        s += "spam dictionary:   " + str(self.spam_dict) + "\n"
+        s += "temperature:       " + str(self.temperature) + "K" + "\n"
+        s += "number of runs:    " + str(self.runs) + "\n"
+        s += "samples per runs:  " + str(self.samples_per_run) + "\n"
+        s += "laser waist:       " + str(self.laser_waist) + "μm \n"
+        s += "\n" + "Solver Options: \n"+str(self.solver_options)[10:-1]+"\n"
+        return s
+
+    @property
+    def noise(self) -> List[str]:
+        return self._noise
+
+    @noise.setter
+    def noise(self, noise_types: List[str]) -> None:
+        self._noise: List[str] = []
+        for noise_type in noise_types:
+            self._add_noise(noise_type)
+
+    def _add_noise(self, noise_type: str) -> None:
+        """Adds a noise model to the SimConfig instance, to be used
+            in Simulation.
+            Args:
+                noise_type (str): Choose among:
+                    'dephasing': random phase (Z) flip
+                    'doppler': Noisy doppler runs
+                    'amplitude': Noisy gaussian beam
+                    'SPAM': SPAM errors. Adds:
+                        eta: Probability of each atom to be badly prepared
+                        epsilon: false positives
+                        epsilon_prime: false negatives
+        """
+        # Check proper input:
+        if noise_type not in noise_dict_set:
+            raise ValueError('Not a valid noise type')
+        self._noise.append(noise_type)
+
+    @property
+    def spam_dict(self) -> dict[str, float]:
+        return self._spam_dict
+
+    def set_spam(self, **values: float) -> None:
```

Comment:
Instead of `**values`, perhaps it would be more informative to specify that they are `eta`, `epsilon` and `epsilon_prime`. Use `Optional` type hints to let the user know they don't have to specify all three